src/error_maps.jl

```diff
@@ -0,0 +1,279 @@
+
+"""
+    ErrorMap
+
+A nonlinear mapping between the space of unit quaternions and three-dimensional
+rotation errors.
+
+These mappings are extremely useful for converting from globally nonsingular 3D rotation
+representations such as `UnitQuaternion` or `RotMatrix3` to a three-parameter error that can
+be efficiently used in gradient-based optimization methods that optimize deviations about
+a current iterate using first or second-order information.
+
+# Usage
+    errmap(v::AbstractVector)  # "forward" map from a 3D error to a `UnitQuaternion`
+    errmap(R::Rotation)        # "inverse" map from a rotation (via `UnitQuaternion`) to a 3D error
+
+where `errmap <: ErrorMap`
+
+# Implemented Maps
+- `CayleyMap`:  Uses `RodriguesParam` as the error representation (default).
+    Goes singular at 180° and does not have a sign ambiguity.
+- `ExponentialMap`: Uses the canonical exponential map from Lie Group theory. Computationally
+    expensive to compute. Exhibits kinematic singularities.
+- `MRPMap`: Uses a scaled `MRP` as the error representation. Singular at 360° but has
+    a sign ambiguity (with the "shadow" MRP set).
+- `QuatVecMap`: Uses the vector part of the quaternions. Cheapest map to compute, but
+    goes singular at 180° and suffers from sign ambiguity.
+- `IdentityMap`: Maps values through directly. Only works with three-parameter rotation
+    representations with the following methods: `R(::SVector{3})` and `SVector(::R)::SVector{3}`
+"""
+abstract type ErrorMap end
+struct CayleyMap <: ErrorMap end
+struct ExponentialMap <: ErrorMap end
+struct MRPMap <: ErrorMap end
+struct QuatVecMap <: ErrorMap end
+struct IdentityMap <: ErrorMap end
+
+const DEFAULT_EMAP = CayleyMap()
```

Comment:
This is no longer used I think?

Reply:
Removed, since the defaults are now baked into the \ominus symbol.

Reply:
Neato!